### ISSUE TITLE
Test failures in Linux/x64/pri1r2r

### ISSUE THREAD

**Issue by pgavlin** [dotnet/coreclr#12643]
e.g. https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_ubuntu_pri1r2r_tst/2165/

A binary search indicates that these failures came in with the changes in https://github.com/dotnet/coreclr/pull/11962.

@ruben-ayrapetyan @janvorli can you take a look?

**Comment by ruben-ayrapetyan** [comment]
@pgavlin, I'll take a look today or early next week.

**Comment by ruben-ayrapetyan** [comment]
Minimal test case:
```
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System.Runtime.InteropServices;

[StructLayout(LayoutKind.Sequential)]
public class GenBase<T>
{
        public T Fld10;
}

[StructLayout(LayoutKind.Sequential)]
public class GenInt : GenBase<int>
{
}
```

Seems that failure is cause by incorrect copy of `FieldMarshaler`s in `EEClassLayoutInfo::CollectLayoutFieldMetadataThrowing`.

The function uses `memcpy` to copy the structures:
```
            memcpyNoGCRefs( ((BYTE*)pEEClassLayoutInfoOut->GetFieldMarshalers()) + MAXFIELDMARSHALERSIZE*numChildCTMFields,
                            pParentLayoutInfo->GetFieldMarshalers(),
                            MAXFIELDMARSHALERSIZE * (pParentLayoutInfo->m_numCTMFields) );
```

This became incorrect after the #11962, because after the change the `FieldMarshaler` structure contains relative pointer `RelativeFixupPointer<PTR_FieldDesc> m_pFD`, and so this should have been changed in the pull request.

I plan to open pull request with fix soon.

**Comment by ruben-ayrapetyan** [comment]
@pgavlin, I checked the #12673 fix (b4131eab7371b3a34c76349771e8766037754bd9) for the failed tests from https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_ubuntu_pri1r2r_tst/2165/ locally (Linux.x64.Debug and Linux.x64.Release) - no fails occured with the fix.
